src/smart-components/group/policy/policy-actions/edit-policy-roles.js

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { withRouter } from 'react-router-dom';
+import { ActionGroup, Button, FormGroup, Modal, Split, SplitItem, Stack, StackItem } from '@patternfly/react-core';
+import { addNotification } from '@redhat-cloud-services/frontend-components-notifications';
+import { updatePolicy, fetchPolicy } from '../../../../redux/actions/policy-actions';
+import { fetchRoles } from '../../../../redux/actions/role-actions';
+import { PolicyRolesLoader } from '../../../../presentational-components/shared/loader-placeholders';
+import PolicySetRoles from '../../add-group/policy-set-roles';
+import '../../../../App.scss';
+
+const EditPolicyRolesModal = ({
+  history: { push },
+  match: { params: { id }},
+  addNotification,
+  fetchPolicy,
+  updatePolicy,
+  postMethod,
+  isFetching,
+  closeUrl
+}) => {
+  const [ roles, setRoles ] = useState([]);
+  const [ policy, setPolicy ] = useState({});
+  const [ selectedRoles, setSelectedRoles ] = useState({});
+
+  const setPolicyData = (policyData) => {
+    setPolicy(policyData);
+    if (policyData && policyData.roles) {
+      setSelectedRoles(policyData.roles.map(role => ({ value: role.uuid, label: role.name, id: role.uuid })));
+    }
+  };
+
+  const fetchData = () => {
+    fetchRoles().payload.then((data) => setRoles(data));
+    fetchPolicy(id).then((data) => { setPolicyData(data.value); });
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const onSave = () => {
+    if (policy) {
+      const policy_data = {
+        name: policy.name,
+        group: policy.group.uuid,
+        roles: selectedRoles.map(role => role.value)
+      };
+      return postMethod ? updatePolicy(policy.uuid, policy_data).then(() => postMethod()).then(() => push(closeUrl)) :
+        updatePolicy(policy.uuid, policy_data).then(() => push(closeUrl));
+    }
+    else {
```

Comment:
no need fo the else branch. Main branch has return statement.

Reply:
@Hyperkid123  - Updated